packages-next/fields/src/types/image/views/Field.tsx

```diff
@@ -0,0 +1,392 @@
+                  }}
+                >
+                  Cancel
+                </Button>
+              )}
+              {errorMessage ? (
+                <Pill tone="negative" weight="light">
+                  {errorMessage}
+                </Pill>
+              ) : (
+                value.kind === 'upload' && (
+                  <Pill weight="light" tone="positive">
+                    Save to upload this image
+                  </Pill>
+                )
+              )}
+            </Stack>
+          </Stack>
+        )}
+      </Stack>
+    ) : (
+      <Stack gap="small">
+        <Stack css={{ alignItems: 'center' }} gap="small" across>
+          <Button
+            size="small"
+            disabled={onChange === undefined}
+            onClick={() => {
+              inputRef.current?.click();
+            }}
+            tone="positive"
+          >
+            Upload Image
+          </Button>
+          <Button size="small" tone="passive" onClick={toggleSetRefUI}>
+            {'Paste Ref'}
+          </Button>
+          {value.kind === 'remove' && value.previous && (
+            <Button
+              size="small"
+              tone="negative"
+              onClick={() => {
+                setSetRef(false);
+                if (value.previous !== undefined) {
+                  onChange?.(value?.previous);
+                }
+              }}
+            >
+              Undo removal
+            </Button>
+          )}
+          {value.kind === 'remove' &&
+            // NOTE -- UX decision is to not display this, I think it would only be relevant
+            // for deleting uploaded images (and we don't support that yet)
+            // <Pill weight="light" tone="warning">
+            //   Save to remove this image
+            // </Pill>
+            null}
+        </Stack>
+      </Stack>
+    );
+
+  return (
+    <FieldContainer>
+      <FieldLabel>{field.label}</FieldLabel>
+      {onChange && canSetRef ? (
+        <RefView
+          onChange={onRefChange}
+          error={forceValidation && errorMessage ? errorMessage : undefined}
+          onCancel={() => {
+            setSetRef(false);
+            if (value.kind === 'from-server') return;
+            if (previousFile && previousFile.kind === 'from-server') {
+              // if value.kind === 'from-server' save state
+              return onChange?.({ kind: 'remove', previous: previousFile });
+            }
+            return onChange?.({ kind: 'remove' });
+          }}
+        />
+      ) : (
+        imgView
+      )}
+      <input
+        css={{ display: 'none' }}
+        autoComplete="off"
+        autoFocus={autoFocus}
+        ref={inputRef}
+        key={inputKey}
+        name={field.path}
+        onChange={onUploadChange}
+        type="file"
+        disabled={onChange === undefined}
+      />
+    </FieldContainer>
+  );
+}
+
+export function validateRef({ ref }: { ref: string }) {
+  if (!isValidImageRef(ref)) {
```

Comment:
@mitchellhamilton  ended up exporting isValidImageRef from utils-legacy because calling parseIamgeRef from here has a couple of issues: 
* `parseImageRef` throws an error both on incorrect regex and on incorrect extensions. I assume we want this behaviour on the backend but we probably don't want this in our Field.
* `parseImageRef` expects to return an object if successful, in this case all we want is a simple boolean check, exposing the check from utils-legacy seems more semantically correct and less error prone. 

Not married to this idea, I think the first point is the only real problem